lib/modules/platform/github/index.ts

```diff
@@ -1731,6 +1732,28 @@ export async function updatePr({
   }
 }
 
+export async function reattemptPlatformAutomerge({
+  number,
+  platformOptions,
+}: ReattemptPlatformAutomergeConfig): Promise<void> {
+  try {
+    const { body: ghPr } = await githubApi.getJson<GhRestPr>(
+      `repos/${config.parentRepo ?? config.repository}/pulls/${number}`,
+    );
+    const result = coerceRestPr(ghPr);
+    const { node_id } = result;
+
+    await tryPrAutomerge(number, node_id, platformOptions);
```

Comment:
won't this merge a PR which doesn't use platform automerge? we don't want that on this stage.

Reply:
My understanding is: no. `tryPrAutomerge` contains all the checks for whether platform automerge should be used, so it'll do nothing as appropriate

Reply:
The problem might be that you're doing an extra API call every rebase even if no automerge platform automerge is enabled

Reply:
I think this check could be done up in the application logic, e.g. if config.automerge and config.platformAutomerge and platform.reattemptPlatformAutomerge then.....

Reply:
Ah, I thought the API call was cached at this point so that it wouldn't fire again.

I've added the checks from `tryPrAutomerge` up higher in the last push. I didn't add a check for `config.automerge` because I saw `createPr` doesn't do that, and I wanted this feature to have the same behavior, but please let me know if you'd like that included!

https://github.com/renovatebot/renovate/blob/main/lib/modules/platform/github/index.ts#L1686

Reply:
Looking one level deeper, I see `getPlatformPrOptions` accounts for `config.automerge` (among other things) when determining its `usePlatformAutomerge` value, so this should be fully covered.

https://github.com/renovatebot/renovate/blob/0b3b56adb555ae993a8f89ea721c5dfa36c4b259/lib/workers/repository/update/pr/index.ts#L50-L54